kiwi_keg/obs_service/compose_kiwi_description.py

```diff
@@ -252,9 +262,23 @@ def main() -> None:
                 )
             image_version = get_image_version(os.path.join(args['--outdir'], 'config.kiwi'))
 
-        for source_log in glob.glob(os.path.join(args['--outdir'], 'log_sources*')):
-            flavor = source_log[len(os.path.join(args['--outdir'], 'log_sources')) + 1:]
-            generate_changelog(source_log, args['--outdir'], flavor, image_version, rev_args)
+        for source_log, flavor in get_log_sources(os.path.join(args['--outdir'])):
+            changes_filename = f'{flavor}{"." if flavor else ""}changes.yaml'
+            changes_path = os.path.join(args['--outdir'], changes_filename)
+            have_changes |= generate_changelog(source_log, changes_path, image_version, rev_args)
```

Comment:
I don't understand that bit, so `generate_changelog` returns `return result.returncode == 0` (line 170)
which means the method/command run inside the method did not produce any errors, right ?
My question is: if no errors == there are changes in the changelog ?

Reply:
Basically yes. `generate_recipes_changelog` returns 2 in case there wasn't an error but no new changes either. The `generate_changelog` functions abort on error and otherwise translates the exist status to True/False.

Reply:
Ack, that makes sense to me now
Maybe a comment to clarify that ?

Reply:
Not a bad idea. :)